src/NServiceBus.NHibernate.Tests/SynchronizedStorage/NHibernateAmbientTransactionSynchronizedStorageSessionTests.cs

```diff
@@ -20,26 +21,28 @@ public async Task It_invokes_callbacks_when_session_is_completed()
                 transportTransaction.Set(Transaction.Current);
 
                 var callbackInvoked = 0;
-                var adapter = new NHibernateSynchronizedStorageAdapter(SessionFactory, null);
 
-                using (var storageSession = await adapter.TryAdapt(transportTransaction, new ContextBag()))
+                //The open method creates the NHibernateLazyNativeTransactionSynchronizedStorageSession
+                var syncSession = new NHibernateSynchronizedStorageSession(new SessionFactoryHolder(SessionFactory));
+                var success = await syncSession.TryOpen(transportTransaction, new ContextBag());
+                Assert.IsTrue(success);
```

Comment:
I'm not a big fan of mixing Assertions with Arrange. How about moving the assertion down?

Reply:
No point in moving it down because it is used right after this line. I wanted to ensure that the test fails here if `TryOpen` returns false. Without this one, the subsequent calls to `Session` would throw not-so-nice exceptions.